src/Repository/ProductCommentRepository.php

```diff
@@ -416,4 +489,32 @@ private function getLastComment(array $criteria)
 
         return empty($comments) ? [] : $comments[0];
     }
+
+    /**
+     * @param int $langId
+     * @param int $shopId
+     *
+     * @return array
+     */
+    public function getReportedComments($langId, $shopId)
+    {
+        /** @var QueryBuilder $qb */
+        $qb = $this->connection->createQueryBuilder();
+        $qb
+            ->select('pc.`id_product_comment`, pc.`id_product`, pc.`content`, pc.`grade`, pc.`date_add`, pl.`name`, pc.`title`
+            , IF(c.id_customer, CONCAT(c.`firstname`, \' \',  c.`lastname`), pc.customer_name) customer_name')
+            ->distinct()
```

Comment:
it seems that `distinct` is not available in PrestaShop 1.7.6, at least, it is what PHPStan returns. Two options I see:
- bump compat to 1.7.7
- change distinct to grouping by a field you want to have without duplicates

Reply:
I'd prefer bump compat to PS 1.7.7 and PHP 7.1.3. Progi1984 even dropped the whole PS 1.7 branch at https://github.com/PrestaShop/blockwishlist/pull/209/files

Reply:
Maybe let's do that. We could then think about refactoring the module to use more modern solutions here and there. Let's bump it to v8. If you change to v8 you need also to introduce a new major version of the module. It will also require removing the tests from .ci corresponding to the old PS versions :)